File: Katiba55.API/Entities/Project.cs
namespace Katiba55.API.Entities
{
    public class Project: BaseEntity
    {
        public string Name { get; set; }
        public string? ExecutingSide { get; set; } 
        public string? BenefitingSide { get; set; } 
        public double? EstimatedCost { get; set; } 
        public double? FinancialAllocation { get; set; }
        public DateTime? EstimatedStartDate { get; set; }
        public DateTime? EstimatedEndDate { get; set; }
        public DateTime? ActualStartDate { get; set; }
        public DateTime? ActualEndDate { get; set; }
        public string? Address { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }

        public double? ExecutionPercent { get; set; }
        public DateTime? ExecutionDate { get; set; }
        public ExecutionStatus ExecutionStatus { get; set; }

        public string? Notes { get; set; }

        public int SupervisorId { get; set; }
        public Officer Supervisor { get; set; }

        public ICollection<Media>? Medias { get; set; }
        public ICollection<Work>? Works { get; set; }
        public ICollection<Item>? Items { get; set; }
        public ICollection<ProjectExecutionHistory>? ExecutionHistories { get; set; }
    }
}

File: Katiba55.Client/src/app/models/projects/project.ts
import { ExecutionStatus } from "../../enums/execution-status.enum"

export interface Project {
    id: number
    name: string
    executingSide?: string
    benefitingSide?: string
    estimatedCost?: number
    financialAllocation?: number
    estimatedStartDate?: Date
    estimatedEndDate?: Date
    actualStartDate?: Date
    actualEndDate?: Date
    address?: string
    latitude?: number
    longitude?: number
    executionStatus: ExecutionStatus
    executionPercent?: number
    executionDate?: Date
    supervisorId: number
    notes?: string
}
